regolith/tools.py

```diff
@@ -2102,6 +2105,170 @@ def google_cal_auth_flow():
+                                        "for details.")
+                                    return False
+                            else:
+                                print(
+                                    "WARNING: The request parameters may not be defined. "
+                                    "If you would like regolith to automatically create a repository in GitHub/GitLab, "
+                                    "please add repository information in regolithrc.json. See regolith documentation "
+                                    "for details.")
+                                return False
+                        else:
+                            print(
+                                "WARNING: The request parameters may not be defined. "
+                                "If you would like regolith to automatically create a repository in GitHub/GitLab, "
+                                "please add repository information in regolithrc.json. See regolith documentation "
+                                "for details.")
+                            return False
+                    else:
+                        print(
+                            "WARNING: The request parameters may not be defined. "
+                            "If you would like regolith to automatically create a repository in GitHub/GitLab, "
+                            "please add repository information in regolithrc.json. See regolith documentation "
+                            "for details.")
+                        return False
+                    if 'url' in rc.repos[target_repo]:
+                        if rc.repos[target_repo]['url']:
+                            url = urlparse(rc.repos[target_repo]['url'])
+                            if url.scheme and url.netloc and url.path:
+                                return True
+                            else:
+                                print(
+                                    "WARNING: The request url may not be valid. "
+                                    "If you would like regolith to automatically create a repository in GitHub/GitLab, "
+                                    "please add repository information in regolithrc.json. See regolith documentation "
+                                    "for details.")
+                                return False
+                        else:
+                            print(
+                                "WARNING: The request url may not be defined. "
+                                "If you would like regolith to automatically create a repository in GitHub/GitLab, "
+                                "please add repository information in regolithrc.json. See regolith documentation "
+                                "for details.")
+                            return False
+                    else:
+                        print(
```

Comment:
whenever cut and paste of code is happening it is time to define a variable
```
message_url_not_defined = ""WARNING: The request url may not be defined. "
                            "If you would like regolith to automatically create a repository in GitHub/GitLab, "
                            "please add repository information in regolithrc.json. See regolith documentation "
                            "for details.")
```
and so on.